R/getSurvival.R

```diff
@@ -0,0 +1,288 @@
+#' representing time to event or follow-up time. Must be numeric.
+#'
+#' @param status.col \code{Character scalar}. Column name in \code{colData(x)} 
+#' representing event occurrence. Accepts numeric (0/1) or logical (FALSE/TRUE) 
+#' values.
+#'
+#' @param covar.cols \code{Character vector}. Optional. Specifies covariate 
+#' columns in \code{colData(x)} to adjust for in the survival model. 
+#' (Default: \code{NULL})
+#' 
+#' @param ... additional arguments.
+#' \itemize{
+#'   \item \code{penalized}: \code{Logical}. If TRUE, fit penalized Cox 
+#'   regression using \code{glmnet}. If FALSE, fit standard Cox model using 
+#'   \code{survival::coxph}. (Default: \code{TRUE})
+#'
+#'   \item \code{lambda}: \code{Character or numeric}. Penalization parameter 
+#'   passed to \code{\link[glmnet]{cv.glmnet}}. Use \code{"lambda.1se"}, 
+#'   \code{"lambda.min"}, or a numeric value. (Default: \code{"lambda.1se"})
+#'
+#'   \item \code{alpha}: \code{Numeric scalar}. Elastic net mixing parameter: 
+#'   1 = Lasso, 0 = Ridge. (Default: \code{0.9})
+#'
+#'   \item \code{nfolds}: \code{Integer scalar}. Number of cross-validation 
+#'   folds for \code{cv.glmnet}. (Default: \code{10})
+#' 
+#'   \item \code{nvar}: \code{Integer scalar}. Optional. Maximum number of 
+#'   variables (log-ratios) to include in the model. (Default: \code{NULL})
+#' 
+#'   \item \code{coef_threshold}: \code{Numeric scalar}. Minimum absolute value 
+#'   for a coefficient to be included in the final model. (Default: \code{0})
+#' }  
+#' 
+#' @inheritParams addBaselineDivergence
+#'
+#' @return A list with model summaries:
+#' \itemize{
+#'   \item \code{coefficients}: estimated model coefficients
+#'   \item \code{selected.features}: nonzero features in penalized model
+#'   \item \code{risk.scores}: predicted risk scores
+#'   \item \code{cindex.apparent}: apparent concordance index
+#'   \item \code{cv.cindex.mean}: mean cross-validated C-index (if penalized)
+#'   \item \code{cv.cindex.sd}: SD of cross-validated C-index (if penalized)
+#'   \item \code{fit}: fitted model object (coxph or cv.glmnet)
+#' }
+#'
+#' @seealso \code{\link[coda4microbiome]{coda_coxnet}}, 
+#' \code{\link[glmnet]{cv.glmnet}}
+#' 
+#' @references
+#' Meritxell Pujolassos , Antoni Susín , M.Luz Calle (2024). 
+#' \emph{Microbiome compositional data analysis for survival studies }
+#' NAR Genomics and Bioinformatics, 6(2), lqae038. 
+#' \doi{10.1093/nargab/lqae038}
+#'
+#' @examples
+#' # data(SurvivalData)
+#' # tse <- SurvivalData
+#' # getSurvival(tse, time.col = "T1Dweek", status.col = "T1D", 
+#' # covar.cols = c("Sex", "Antibiotics"))
+#'
+NULL
+
+#' @rdname getSurvival
+#' @export
+#' @importFrom survival Surv coxph
+#' @importFrom glmnet cv.glmnet glmnet
+setMethod("getSurvival", signature(x = "SummarizedExperiment"),
+    function(
+        x,
+        time.col,
+        status.col,
+        assay.type = "counts",
+        covar.cols = NULL,
```

Comment:
Check argument names that we are already using. I think we are using `col.var` which would suit also here

Reply:
Could also be defined as a formula